flare/dft_interface/qe_util.py

```diff
@@ -9,23 +9,32 @@
 def run_dft(qe_input, structure, dft_loc):
     run_qe_path = qe_input
     edit_dft_input_positions(run_qe_path, structure)
-    qe_command = '{0} < {1} > {2}'.format(dft_loc, run_qe_path,
-                                                 'pwscf.out')
+    qe_command = '{0} < {1} > {2}'.format(dft_loc, run_qe_path, 'pwscf.out')
     call(qe_command, shell=True)
 
     return parse_dft_forces('pwscf.out')
 
 
-def run_dft_par(qe_input, structure, dft_loc, no_cpus):
-    run_qe_path = qe_input
-    edit_dft_input_positions(run_qe_path, structure)
-    qe_command = \
-        'mpirun -np {0} {1} < {2} > {3}'.format(no_cpus, dft_loc, run_qe_path,
-                                                'pwscf.out')
+def run_dft_par(qe_input, structure, dft_loc, no_cpus=1, dft_out='pwscf.out',
+                npool=None):
+    newfilename = edit_dft_input_positions(qe_input, structure)
 
-    call(qe_command, shell=True)
+    if npool is None:
+        dft_command = \
+            '{} -i {} > {}'.format(dft_loc, newfilename, dft_out)
+    else:
+        dft_command = \
+            '{} -npool {} -i {} > {}'.format(dft_loc, npool, newfilename,
+                                             dft_out)
 
-    return parse_dft_forces('pwscf.out')
+    if (no_cpus > 1):
+        # dft_command = 'mpirun -np {} {}'.format(no_cpus, dft_command)
+        dft_command = 'srun -n {} --mpi=pmi2 {}'.format(no_cpus, dft_command)
```

Comment:
This line is the culprit for unit test failure. Should leave it to the user to decide.